src/databricks/labs/ucx/workspace_access/clusters.py

```diff
@@ -0,0 +1,117 @@
+import logging
+
+from databricks.labs.blueprint.installation import Installation
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import InvalidParameterValue
+from databricks.sdk.service.compute import ClusterDetails, DataSecurityMode
+
+logger = logging.getLogger(__name__)
+
+
+class ClusterAccess:
+    def __init__(self, installation: Installation, ws: WorkspaceClient, prompts: Prompts):
+        self._ws = ws
+        self._prompts = prompts
+        self._installation = installation
+
+    def list_cluster(self):
+        cluster_list = {}
+        clusters = self._ws.clusters.list()
+        for cluster in clusters:
+            if cluster.cluster_source is not None and cluster.cluster_source.name != "JOB":
+                cluster_list[cluster.cluster_name] = cluster.cluster_id
+        return cluster_list
+
+    def _get_cluster_id(self, cluster_id: str):
+        if cluster_id != "<ALL>":
+            return [x.strip() for x in cluster_id.split(",")]
+        cluster_list = []
+        clusters = self._ws.clusters.list()
+        for cluster in clusters:
+            if cluster.cluster_source is not None and cluster.cluster_source.name != "JOB":
+                cluster_list.append(cluster.cluster_id)
+        return cluster_list
+
+    def _get_access_mode(self, access_mode: str):
+        if access_mode in {"LEGACY_SINGLE_USER", "SINGLE_USER"}:
+            return DataSecurityMode.SINGLE_USER
+        return DataSecurityMode.USER_ISOLATION
+
+    def map_cluster_to_uc(self, cluster_id: str):
+
+        cluster_id_list = self._get_cluster_id(cluster_id)
+        spark_version = self._ws.clusters.select_spark_version(latest=True)
+        for cluster in cluster_id_list:
+            try:
+                cluster_details = self._ws.clusters.get(cluster)
+                if cluster_details.data_security_mode is None:
+                    logger.info(f"Data security Mode is None. Skipping the remapping for the cluster: {cluster}")
+                    continue
+                access_mode = self._get_access_mode(cluster_details.data_security_mode.name)
+                self._installation.save(cluster_details, filename=f'backup/clusters/{cluster_details.cluster_id}.json')
+                self._ws.clusters.edit(
```

Comment:
Is there a way to make this resilient to cluster API changes? The Cluster UI and options are constantly changing, how can this code just focus on the specifics of credentials, spark configs and data security mode and pass through all the other configuration stuff. Will this code break frequently? [I don't know]